### PR TITLE
chore: replace deprecated querystring package with URLSearchParams API closes #3562

### LINKED ISSUE #3562: chore: remove deprecated querystring package with URLSearchParams
**Describe the bug**

The [querystring](https://www.npmjs.com/package/querystring) package has been deprecated now and should be replaced by URLSearchParams API as recommended in its documentation.

It's used in 3 files: `functions-templates/javascript/google-analytics/{{name}}.js`, `functions/form-submissions-handler` and `utils/gh-auth`.

### PR DESCRIPTION
**Pull request by tinfoil-knight**
#### Summary

Closes #3562 

The querystring package has been replaced with the URLSearchParams API since querystring has been deprecated.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [x] Make sure the status checks below are successful ✅

### CODE REVIEW

File: src/lib/functions/form-submissions-handler.js [hunk 1]
```diff
@@ -126,6 +126,12 @@ const createFormSubmissionHandler = function ({ functionsRegistry, siteUrl }) {
   }
 }
 
+// note: use Object.fromEntries post Nodejs 12.0.0
+const getObjectFromParams = (urlParams) => {
+  const entries = [...urlParams]
+  return entries.reduce((prev, [key, value]) => ({ ...prev, [key]: value }), {})
+}
+
```
Comment (lukasholzer) [inline src/lib/functions/form-submissions-handler.js:134]: NIT: Maybe you could mark the function as deprecated as there is already an internal issue to drop the support of Nodejs < 12.0.0

```suggestion
/**
 * Converts URLSearchParams to an object
 * @deprecated can be replaced with `Object.fromEntries` once the support for older Node.js Versions than `12.0.0` is dropped.
 * @param {URLSearchParams} urlParams 
 * @returns {Record<string, any>}
 */
?**
const getObjectFromParams = (urlParams) => {
  const entries = [...urlParams]
  return entries.reduce((prev, [key, value]) => ({ ...prev, [key]: value }), {})
}

```

**Review by lukasholzer — DISMISSED** [review]
Thank you very much for taking the time and cleaning up the `querystring` package 🐳

File: src/lib/functions/form-submissions-handler.js (hunk 1 above)
Comment (tinfoil-knight) [reply src/lib/functions/form-submissions-handler.js]: I've added the suggestion.

File: src/lib/functions/form-submissions-handler.js [hunk 2]
```diff
@@ -126,7 +126,13 @@ const createFormSubmissionHandler = function ({ functionsRegistry, siteUrl }) {
...
 }
 
-// note: use Object.fromEntries post Nodejs 12.0.0
+/**
+ * Converts URLSearchParams to an object
+ * @deprecated can be replaced with `Object.fromEntries` once the support for older Node.js Versions than `12.0.0` is dropped.
+ * @param {URLSearchParams} urlParams 
+ * @returns {Record<string, any>}
+ */
+?**
```
Comment (lukasholzer) [inline src/lib/functions/form-submissions-handler.js:135]:
```suggestion
```

Comment (tinfoil-knight) [reply src/lib/functions/form-submissions-handler.js]: Yep. Saw that and removed it in the commit. Thank you.

**Review by lukasholzer — APPROVED** [review]

**Pull request merged**